ledger-core/v2/virtual/object/object.go

```diff
@@ -95,12 +111,45 @@ func (sm *SMObject) Init(ctx smachine.InitializationContext) smachine.StateUpdat
 	if !ctx.Publish(sm.Reference.String(), sdl) {
 		return ctx.Stop()
 	}
-	return ctx.Jump(sm.stepReadyToWork)
+
+	switch sm.initReason {
+	case InitReasonCTConstructor:
+		return ctx.Jump(sm.stepReadyToWork)
+	case InitReasonCTMethod:
+		return ctx.Jump(sm.stepGetObjectState) // TODO PLAT-294 - we will jump to stepWaitState
+	case InitReasonVStateReport:
+		return ctx.Jump(sm.stepWaitState)
+	default:
+		panic("Not implemented")
+	}
+}
+
+// we get CallMethod but we have no object data
+// we need to ask previous executor
+func (sm *SMObject) stepGetObjectState(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	msg := payload.VStateRequest{
+		Callee:           sm.Reference,
+		RequestedContent: payload.RequestLatestDirtyState,
+	}
+
+	goCtx := ctx.GetContext()
+	sm.messageSender.PrepareNotify(ctx, func(svc messagesender.Service) {
```

Comment:
Sure, `PrepareNotify().Send()`is called in context of current SM, but try to avoid direct access to sm memory in all adapter callbacks.
For readability, please, extract `sm.pulseSlot.PulseData().PrevPulseNumber()` to separate variable.
And also i'm not sure that we should use `.PrevPulseNumber()` there, since pulse can be skipped. We need somehow to get previous pulse from storage.

Reply:
fixed